cpp/mg_utility/mg_utils.hpp

```diff
@@ -202,67 +220,159 @@ std::unique_ptr<mg_graph::Graph<TSize>> GetWeightedGraphView(mgp_graph *memgraph
+    mgp_list *subgraph_nodes, mgp_list *subgraph_edges,
+    const mg_graph::GraphType graph_type, const char *weight_property,
+    double default_weight) {
+  auto graph = std::make_unique<mg_graph::Graph<TSize>>();
+
+  /// Getting subgraph node and edge IDs
+
+  std::unordered_set<std::uint64_t> subgraph_node_ids;
+  for (std::size_t i = 0; i < mgp::list_size(subgraph_nodes); i++) {
+    subgraph_node_ids.insert(
+        mgp::vertex_get_id(
+            mgp::value_get_vertex(mgp::list_at(subgraph_nodes, i)))
+            .as_int);
+  }
+
+  std::unordered_set<std::uint64_t> subgraph_edge_ids;
+  for (std::size_t i = 0; i < mgp::list_size(subgraph_edges); i++) {
+    subgraph_edge_ids.insert(
+        mgp::edge_get_id(mgp::value_get_edge(mgp::list_at(subgraph_edges, i)))
+            .as_int);
+  }
+
+  ///
+  /// Mapping Memgraph in-memory vertices into graph view
+  ///
+
+  // Safe vertices iterator creation
+  auto *vertices_it = mgp::graph_iter_vertices(memgraph_graph, memory);
+  mg_utility::OnScopeExit delete_vertices_it(
+      [&vertices_it] { mgp::vertices_iterator_destroy(vertices_it); });
+
+  // Iterate through Memgraph vertices and map them to GraphView
+  for (auto *vertex = mgp::vertices_iterator_get(vertices_it); vertex;
+       vertex = mgp::vertices_iterator_next(vertices_it)) {
+    if (subgraph_node_ids.find(mgp::vertex_get_id(vertex).as_int) ==
+        subgraph_node_ids.end()) {
+      continue;
+    }
+
+    mg_graph::CreateGraphNode(graph.get(), vertex);
+  }
+  // Destroy iterator before creating a new one in order to avoid memory leakage
+  mgp::vertices_iterator_destroy(vertices_it);
+
+  ///
+  /// Mapping Memgraph in-memory edges into graph view
+  ///
+
+  // Safe vertices iterator creation
+  vertices_it = mgp::graph_iter_vertices(memgraph_graph, memory);
+
+  for (auto *vertex_from = mgp::vertices_iterator_get(vertices_it); vertex_from;
+       vertex_from = mgp::vertices_iterator_next(vertices_it)) {
+    if (subgraph_node_ids.find(mgp::vertex_get_id(vertex_from).as_int) ==
+        subgraph_node_ids.end()) {
+      continue;
+    }
+
+    // Safe edges iterator creation
+    auto *edges_it = mgp::vertex_iter_out_edges(vertex_from, memory);
+    mg_utility::OnScopeExit delete_edges_it(
+        [&edges_it] { mgp::edges_iterator_destroy(edges_it); });
+
+    for (auto *out_edge = mgp::edges_iterator_get(edges_it); out_edge;
+         out_edge = mgp::edges_iterator_next(edges_it)) {
+      if (subgraph_edge_ids.find(mgp::edge_get_id(out_edge).as_int) ==
+          subgraph_edge_ids.end()) {
+        continue;
+      }
+      auto vertex_to = mgp::edge_get_to(out_edge);
+      auto weight = mg_utility::GetNumericProperty(out_edge, weight_property,
+                                                   memory, default_weight);
+      mg_graph::CreateWeightedGraphEdge(graph.get(), vertex_from, vertex_to,
+                                        weight, graph_type);
```

Comment:
Two comments here. Firstly, let's make these methods re-usable rather than copy/pasting from another place. Second thing, the subgraph can be defined only by using a list of edges, no need to send the list of nodes because that can be extracted from the list of edges.

Reply:
Fully agree about your first point. Regarding the second, using only the edges doesn’t work when the subgraph contains isolated nodes, and makes the function inconsistent with existing ones like `color_subgraph`.

Reply:
Added a reusable method for all currently used graph views.